tests/unit/lms/resources/lti_launch_test.py

```diff
@@ -27,25 +26,18 @@ def test_it_allows_the_correct_users_to_launch_LTI_assignments(
 
 
 class TestHGroup:
-    @pytest.mark.parametrize(
-        "context_title,expected_group_name",
-        (
-            ("Test Course", "Test Course"),
-            (" Test Course ", "Test Course"),
-            ("Test   Course", "Test   Course"),
-            ("Object Oriented Polymorphism 101", "Object Oriented Polymorp…"),
-            ("  Object Oriented Polymorphism 101  ", "Object Oriented Polymorp…"),
-        ),
-    )
-    def test_it(self, lti_launch, context_title, expected_group_name, pyramid_request):
+    def test_it(self, lti_launch, pyramid_request, HGroup, h_group_name):
         pyramid_request.parsed_params = {
             "context_id": "test_context_id",
-            "context_title": context_title,
+            "context_title": "test_context_title",
             "tool_consumer_instance_guid": "test_tool_consumer_instance_guid",
         }
 
-        assert lti_launch.h_group == HGroup(
-            expected_group_name, "d55a3c86dd79d390ec8dc6a8096d0943044ea268",
+        assert lti_launch.h_group == HGroup.return_value
+        h_group_name.assert_called_once_with("test_context_title")
+        HGroup.assert_called_once_with(
+            name=h_group_name.return_value,
+            authority_provided_id="d55a3c86dd79d390ec8dc6a8096d0943044ea268",
         )
```

Comment:
Caused some good separation of concerns here. Just FYI, I think for small pure pythons stuff like this I'd be equally happy to black box test it too.